src/components/App/ExpressRecorder.tsx

```diff
@@ -161,8 +161,12 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
      * upload the latest recording to Kaltura
      */
     upload = () => {
-        const { doRecording, recordedBlobs } = this.state;
-        if (!doRecording && recordedBlobs.length > 0 && !this.uploadedOnce) {
+        const { doRecording, recordedBlobs, screenRecordedBlobs } = this.state;
+        if (
+            !doRecording &&
+            (recordedBlobs.length || screenRecordedBlobs.length) &&
+            !this.uploadedOnce
+        ) {
```

Comment:
why is it better to write multiple "if"?
In my why you have all info in one place

Reply:
we should not get used to this writing approach:
1. cos it's hard to maintain and refactor with ease.
2. cos it's not clear when it gets longer.
3. cos it's hard to debug from local and production.
4. we should strive to code from top to bottom - and not from left to right. I prefer more amount of code lines than wide code which is hard to read.
5. moreover, we should try to avoid `if` conditions as much as possible and map our data to relevant components and/or data structures.

Reply:
ok, will do